### PR TITLE
Splitting

### PR DESCRIPTION
**Pull request by zyndaras**

### CODE REVIEW

**Comment by zyndaras** [comment]
@daves125125

File: src/main/java/com/opinari/otf/configuration/JourneyConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoClientFactoryBean;
+import org.springframework.data.mongodb.core.SimpleMongoDbFactory;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+/**
+ * @author David Hancock
+ */
+@org.springframework.context.annotation.Configuration
```
Comment (daves125125) [inline src/main/java/com/opinari/otf/configuration/JourneyConfiguration.java]: dont need to use fully qualifed package name here now

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        mavenBom 'org.springframework.data:spring-data-releasetrain:Hopper-SR1'
+    }
+}
+
+dependencies {
+
+    compile 'org.springframework.boot:spring-boot-starter-web'
+    compile 'org.springframework.data:spring-data-mongodb'
+
+    compile 'org.springframework.security.oauth:spring-security-oauth2:2.0.8.RELEASE'
```
Comment (daves125125) [inline build.gradle]: we dont use either of these spring security / jwt libs in this service atm, it lives in user?

File: src/main/java/com/opinari/otf/domain/option/OptionType.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.opinari.otf.domain.option;
+
+/**
+ * @author Ivan Jimenez Diez
+ */
+public enum OptionType {
```
Comment (daves125125) [inline src/main/java/com/opinari/otf/domain/option/OptionType.java]: Do we use optionType in journey?

File: src/main/java/com/opinari/otf/exception/GlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * @author David Hancock
+ */
+@ControllerAdvice
+public class GlobalExceptionHandler {
```
Comment (daves125125) [inline src/main/java/com/opinari/otf/exception/GlobalExceptionHandler.java]: Need to think about these common classes too... duplication will become irritating when we have lots of microservices, but so will creating jar deps

File: src/main/java/com/opinari/otf/user/service/DefaultUserService.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    public UserClaim authenticate(String token) {
+
+        HttpHeaders requestHeaders = new HttpHeaders();
+        requestHeaders.set("Authorized", token);
+        HttpEntity requestEntity = new HttpEntity(requestHeaders);
+
+        URI uri = UriComponentsBuilder.fromHttpUrl(userHost + USER_AUTHENTICATION_ENDPOINT)
+                .build().encode().toUri();
```
Comment (daves125125) [inline src/main/java/com/opinari/otf/user/service/DefaultUserService.java]: probably fit on one line?

File: src/main/java/com/opinari/otf/OnTheFenceJourneyApplication.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.actuate.autoconfigure.ManagementWebSecurityAutoConfiguration;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.security.SecurityAutoConfiguration;
+
+/**
+ * @author David Hancock
+ */
+@SpringBootApplication(exclude = {
+        SecurityAutoConfiguration.class,
```
Comment (daves125125) [inline src/main/java/com/opinari/otf/OnTheFenceJourneyApplication.java]: Dont need these excludes if we remove the spring security deps.

We might need to bring auth / auth back and use spring security, but not sure when / if that will happen so remove in meantime probably

File: src/main/java/com/opinari/otf/configuration/SwaggerConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+/**
+ * @author David Hancock
+ */
+// TODO We'll be more stringent on this as we get more mature, i.e. creating and activating spring profiles per env.
+@Profile("default")
+@EnableSwagger2
+@Configuration
+public class SwaggerConfiguration {
```
Comment (daves125125) [inline src/main/java/com/opinari/otf/configuration/SwaggerConfiguration.java]: i guess we can keep this chap for now, might be useful to see our api - although swagger will be harder to use now we've fragemented our services

File: src/test/java/com/opinari/otf/MongoDBConnectionIT.java [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
+public class MongoDBConnectionIT {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MongoDBConnectionIT.class);
+
+    @Configuration
+    @EnableAutoConfiguration(exclude = {
+            SecurityAutoConfiguration.class,
+            ManagementWebSecurityAutoConfiguration.class})
```
Comment (daves125125) [inline src/test/java/com/opinari/otf/MongoDBConnectionIT.java]: same as above

**Pull request merged**